lib/Raven/Client.php

```diff
@@ -233,35 +238,49 @@ class Client
      */
     protected $_shutdown_function_has_been_set;
 
-    public function __construct($options_or_dsn = null, $options = [])
+    /**
+     * @var Configuration The client configuration
+     */
+    protected $config;
+
+    /**
+     * Constructor.
+     *
+     * @param Configuration $config The client configuration
+     */
+    public function __construct(Configuration $config)
```

Comment:
No. Never ever. Just no. 

Create method
```php
/**
 * @return Client
 */
function getClient(Configuration $config) {
    // @todo magic...

   return $object;
}
```

Constuctor must be as simple as possible. Only `new \Raven\Client(['key' => 'value']);`